src/main/java/org/symphonyoss/symphony/messageml/elements/TableSelect.java

```diff
@@ -0,0 +1,178 @@
+
+  public static final Set<String> VALID_TYPES = new HashSet<>(Arrays.asList("button", "checkbox"));
+
+  public static final Set<String> VALID_POSITIONS = new HashSet<>(Arrays.asList("left", "right"));
+
+  private final static String LEFT = "left";
+  private final static String RIGHT = "right";
+  private final static String BUTTON = "button";
+  private final static String CHECKBOX = "checkbox";
+
+  public TableSelect(Element parent) {
+    super(parent, MESSAGEML_TAG);
+
+    setAttribute(HEADER_TEXT_ATTR, "Select");
+    setAttribute(BUTTON_TEXT_ATTR, "SELECT");
+  }
+
+
+  @Override
+  public void validate() throws InvalidInputException {
+    super.validate();
+
+    String name = getAttribute(NAME_ATTR);
+
+    if (name == null || name.trim().isEmpty()) {
+      throw new InvalidInputException("The attribute \"name\" is required");
+    }
+
+    String type = getAttribute(TYPE_ATTR);
+
+    if (type == null) {
+      throw new InvalidInputException("The attribute \"type\" is required");
+    }
+
+    if (!VALID_TYPES.contains(type)) {
+      throw new InvalidInputException("Attribute \"type\" must be \"button\" or \"checkbox\"");
+    }
+
+
+    String position = getAttribute(POSITION_ATTR);
+
+
+    if (position == null) {
+      throw new InvalidInputException("The attribute \"position\" is required");
+    }
+
+    if (!VALID_POSITIONS.contains(position)) {
+      throw new InvalidInputException("Attribute \"position\" must be \"left\" or \"right\"");
+    }
+
+    assertContentModel(Arrays.asList(TableHeader.class, TableBody.class, TableFooter.class, TableRow.class));
+  }
+
+  @Override
+  protected void buildAttribute(org.w3c.dom.Node item) throws InvalidInputException {
+    switch (item.getNodeName()) {
+
+      case NAME_ATTR:
+        setAttribute(NAME_ATTR, getStringAttribute(item));
+        break;
+
+      case TYPE_ATTR:
+        setAttribute(TYPE_ATTR, getStringAttribute(item));
+        break;
+
+      case HEADER_TEXT_ATTR:
+        setAttribute(HEADER_TEXT_ATTR, getStringAttribute(item));
+        break;
+
+      case BUTTON_TEXT_ATTR:
+        setAttribute(BUTTON_TEXT_ATTR, getStringAttribute(item));
+        break;
+
+      case POSITION_ATTR:
+        setAttribute(POSITION_ATTR, getStringAttribute(item));
+        break;
+
+      default:
+        throw new InvalidInputException("Attribute \"" + item.getNodeName()
+            + "\" is not allowed in \"" + getMessageMLTag() + "\"");
+    }
+  }
+
+  @Override
+  public void asPresentationML(XmlPrintStream out) {
+    out.openElement(PRESENTATIONML_TABLE_TAG);
+    for (Element child : getChildren()) {
+      child.asPresentationML(out);
+    }
+    out.closeElement(); // Closing table
+  }
+
+  @Override
+  public Node asMarkdown() {
+    return new TableSelectNode();
+  }
+
+  public String getName() {
+    return getAttribute(NAME_ATTR);
+  }
+
+  public String getPosition() {
+    return getAttribute(POSITION_ATTR);
+  }
+
+  public String getType() {
```

Comment:
Why do you need this one if you have the "isButton" and "isCheckbox" methods?

Reply:
Removed